src/module.c

```diff
@@ -8805,6 +8805,72 @@ int RM_ACLAddLogEntry(RedisModuleCtx *ctx, RedisModuleUser *user, RedisModuleStr
     return REDISMODULE_OK;
 }
 
+/* This is to prevent a coupling between redis internals and modules */
+uint64_t mapInternalScriptFlagsToModuleFlags(uint64_t flags) {
+    uint64_t flags_out = 0;
+
+    if (flags & SCRIPT_FLAG_NO_WRITES) flags_out |= REDISMODULE_SCRIPT_FLAG_NO_WRITES;
+
+    if (flags & SCRIPT_FLAG_ALLOW_OOM) flags_out |= REDISMODULE_SCRIPT_FLAG_ALLOW_OOM;
+
+    if (flags & SCRIPT_FLAG_ALLOW_STALE) flags_out |= REDISMODULE_SCRIPT_FLAG_ALLOW_STALE;
+
+    if (flags & SCRIPT_FLAG_NO_CLUSTER) flags_out |= REDISMODULE_SCRIPT_FLAG_NO_CLUSTER;
+
+    if (flags & SCRIPT_FLAG_EVAL_COMPAT_MODE) flags_out |= REDISMODULE_SCRIPT_FLAG_EVAL_COMPAT_MODE;
+
+    if (flags & SCRIPT_FLAG_ALLOW_CROSS_SLOT) flags_out |= REDISMODULE_SCRIPT_FLAG_ALLOW_CROSS_SLOT;
+
+    return flags_out;
+}
+
+/* parses a script to extract the shebang flags that might be part of it
+ *
+ * Ex: Enables a module to decide if it should RM_Call an eval in an oom situation
+ *
+ * The Flags that can be returned are
+ *
+ * * REDISMODULE_SCRIPT_FLAG_NO_WRITES
+ * * REDISMODULE_SCRIPT_FLAG_ALLOW_OOM
+ * * REDISMODULE_SCRIPT_FLAG_ALLOW_STALE
+ * * REDISMODULE_SCRIPT_FLAG_NO_CLUSTER
+ * * REDISMODULE_SCRIPT_FLAG_EVAL_COMPAT_MODE
+ * * REDISMODULE_SCRIPT_FLAG_ALLOW_CROSS_SLOT
```

Comment:
I wonder if we really need this conversion - why not return a string, and parse it on the module side?

Reply:
i don't understand what you mean by string?  it's a flag set in redis, we would have to create some string on demand.  parsing seems a pain. the reason for the conversion is to enable redis to change while keeping the api consistent.

Reply:
The flags are based on a string representation - see `scripts_flags_def`, but anyway we can give the redis oss committee to decide :)

Reply:
@sjpotter ?